hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1027,15 +1030,22 @@ def _from_type_deferred(thing: Type[Ex]) -> SearchStrategy[Ex]:
         module_prefix = f"{thing.__module__}."
         if not thing_repr.startswith(module_prefix):
             thing_repr = module_prefix + thing_repr
+    try:
+        repr_ = f"from_type({thing_repr})"
```

Comment:
I'd expect the exception to be raised where we compute `thing_repr` above; here it's only ever a string and so I don't think that this part can fail.

Reply:
Doh. Yes of course.